app/js/enhancedDate.js

```diff
@@ -1,5 +1,52 @@
-/* 
- * add your module definition for enhancedDate here
- * assign the module to a global variable
- * /
+var enhancedDate = (function() {    
+    var myDate = new Date();    
+    myDate.setHours(0, 0, 0, 0);
+
+    return {    
+        getDate: function() {   
+            return myDate.toDateString();
+        },
+        setDate: function(newDate) {
+            if (isNaN(Date.parse(newDate))) {
+                alert(newDate + " is not a valid date.");               
+            }
+            else {
+                var d = new Date(newDate);
+                d.setHours(0, 0, 0, 0);
+                myDate = d;             
+            }
+        },
+        getDayName: function() {            
+            var weekday = ["Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"];
+            var dayName =  weekday[myDate.getDay()];           
+            return dayName;
+        },
+        getMonthName: function() {          
+            var month = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+            var monthName =  month[myDate.getMonth()];             
+            return monthName;
+        },
+        isFuture: function()
+        {
+            var d = new Date();
+            d.setHours(0, 0, 0, 0);
+            return (myDate.valueOf() > d.valueOf());             
+        },
+        isToday: function()
+        {
+            var d = new Date();
+            d.setHours(0, 0, 0, 0);
+            return (d.valueOf() === myDate.valueOf());                  
```

Comment:
The valueOf method returns a millisecond value. This likelihood of the two millisecond values being different is very high and isn't sufficient to determine if it's "today". There's a couple of ways to do this, but if you get the year, month, and date of d and compare them to the year, month, and date of myDate, you can determine if it is "today" (all three values have to match.)